### ISSUE TITLE
GetOrSetAsync() not refreshing for expired cache entries

### ISSUE THREAD

**Issue by Miggleness** [TurnerSoftware/CacheTower#195]
### What does the bug do?

GetOrSetAsync() not refreshing for expired cache entries due to current `CacheStack`'s `GetOrSetAsync()` always passing `noExistingValueAvailable=true` when calling `RefreshValueAsync()`.

### How can it be reproduced?

You can reproduce the bug by running this code...

```csharp
[TestMethod]
public async Task GetOrSet_ExpiredCacheHit()
{
	await using var cacheStack = new CacheStack(new[] { new MemoryCacheLayer() }, Array.Empty<ICacheExtension>());
	await cacheStack.SetAsync("GetOrSet_CacheHit", 17, TimeSpan.FromDays(-1));

	var result = await cacheStack.GetOrSetAsync<int>("GetOrSet_CacheHit", (oldValue) =>
	{
		Assert.AreEqual(17, oldValue);
		return Task.FromResult(27);
	}, new CacheSettings(TimeSpan.FromDays(1)));

	Assert.AreEqual(27, result);
}
```

### Environment (if applicable)

- NuGet Package Version: 0.11.5
- .NET Runtime Version: .NET Core 5
- Operating System: Windows

**Comment by Turnerj** [comment]
Hey @Miggleness - thanks for raising this! Looks like this is caused by 67deced316a6ba8af7ff373214de28604361aefb where I was trying to work around a race condition. Will try and work out the best course of action to fix it without re-introducing that race condition.

**Comment by Miggleness** [comment]
I could raise a PR too if you don't already have a fix ready

**Comment by Miggleness** [comment]
cool. i was thinking along the same lines. im not sure sure about the locking part when refreshing

**Comment by Turnerj** [comment]
Thanks again @Miggleness for raising this and providing a simple drop-in test case for the bug - little things like that make my life a lot easier! I've just released 0.11.6 which includes the fix for this bug.
